pandas/tests/series/test_indexing.py

```diff
@@ -287,6 +287,12 @@ def test_getitem_generator(self):
         assert_series_equal(result, expected)
         assert_series_equal(result2, expected)
 
+    def test_type_promotion(self):
+        # GH12599
+        df1 = pd.Series()
+        df1["a"] = pd.Timestamp("2016-01-01")
+        df1["b"] = 3.0
+
```

Comment:
This is the essence of the error (and corrected by the proposed fix). So you would like to have both df1 and df2 copied here from https://github.com/pydata/pandas/issues/12599?